### PR TITLE
Fix: EventProcessors were not dropping events when returning null

### LINKED ISSUE #351: Null safety update prevents processors nulling out events
We use the following processor to stop events being sent to Sentry in debug/development:

```dart
Future<void> initializeSentry(String release, String environment, String dsn,
        {FutureOr<void> Function() runApp, Transport transport}) =>
    Sentry.init(
      (options) => options
        ..dsn = dsn
        ..release = release
        ..environment = environment
        ..transport = transport ?? options.transport
        ..addEventProcessor(_noEventInDebug),
      appRunner: runApp,
    );

SentryEvent _noEventInDebug(SentryEvent event, {dynamic hint}) {
  return isInDebugMode || disableSentry ? null : event;
}
```

Prior to null safety this worked as expected, with this test passing:

```dart

class MockTransport extends Mock implements Transport {}

  test('No sentry events should be sent in debug mode', () async {
    expect(isInDebugMode, isTrue);
    expect(overrideDisableSentry, isNull);

    await initializeSentry('release', 'env', fakeDsn, transport: transport);

    expect(Sentry.isEnabled, isTrue);

    final event = SentryEvent();
    await Sentry.captureEvent(event);

    final callCount = verifyNever(
      transport.send(captureAny),
    ).callCount;

    expect(callCount, 0);
  });
```

However, when upgrading to null safety version of the package, this test now fails.

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
EventProcessors were not dropping events when returning null

## :bulb: Motivation and Context
Fix #351

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 * Bump: sentry-android to v4.3.0 (#343)
 * Fix: Multiple FlutterError.onError calls in FlutterErrorIntegration (#345)
+* Fix: EventProcessors were not dropping events when returning null
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
* Fix: EventProcessors were not dropping events when returning null (#353)
```

**Review by ueman — APPROVED** [review]
LGTM

**Comment by fzyzcjy** [comment]
+1 This is of great significance - otherwise we can never drop any events!

When can we see it released?

**Comment by marandaneto** [comment]
> +1 This is of great significance - otherwise we can never drop any events!
> 
> When can we see it released?

Monday :)

**Pull request merged**